frontend/src/components/PetitionCard.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Dispatch,
+  FormEventHandler,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+import { FaSpinner } from "react-icons/fa";
+import { PetitionDetail, Question } from "../models/Petition";
+import { submitPetitionSignature } from "../utils/backend";
+
+interface Props {
+  className?: string;
+  petition: PetitionDetail;
+  link?: ReactNode;
+}
+
+enum State {
+  Initial,
+  Submitting,
+  Error,
+  Existed,
+  Success,
+}
+
+const SESSION_KEY = "petitionData";
+
+function makeBox(
+  props: Props,
+  children: ReactNode,
+  onsubmit: FormEventHandler<HTMLFormElement> | undefined = undefined
```

Comment:
Meestal wil je ook niet expliciet undefined aan iets assignen. Je kan een vraagteken gebruiken om een argument optioneel te maken:
`onSubmit?: FormEventHandler<HTMLFormElement>`

Reply:
dankje dat is echt veel mooier